spec/MongoTransform.spec.js

```diff
@@ -331,6 +331,41 @@ describe('parseObjectToMongoObjectForCreate', () => {
     expect(output.ts.iso).toEqual('2017-01-18T00:00:00.000Z');
     done();
   });
+
+  it('$regex in $all list', (done) => {
+    var input = {
+      arrayField: {'$all': [{$regex: '^\\Qone\\E'}, {$regex: '^\\Qtwo\\E'}, {$regex: '^\\Qthree\\E'}]},
+    };
+    var outputValue = {
+      arrayField: {'$all': [/^\Qone\E/, /^\Qtwo\E/, /^\Qthree\E/]},
+    };
+
+    var output = transform.transformWhere(null, input);
+    jequal(outputValue.arrayField, output.arrayField);
+    done();
+  });
+
+  it('$regex in $all list must be { $regex: "string" }', (done) => {
+    var input = {
+      arrayField: {'$all': [{$regex: 1}]},
+    };
+
+    expect(() => {
+      transform.transformWhere(null, input)
+    }).toThrow();
+    done();
+  });
+
+  it('all values in $all must be $regex (start with string) or non $regex (start with string)', (done) => {
```

Comment:
Thanks for adding the unit tests here. I know you are checking for isArray. I'm curious to find out what would happen if I put

$all: false
$all: 1234
$all: undefined
$all: null
$all: {}